src/MinterGateway.sol

```diff
@@ -866,13 +873,20 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
      * @param newTimestamp_ The timestamp of the collateral update.
      */
     function _updateCollateral(address minter_, uint240 amount_, uint40 newTimestamp_) internal {
-        uint40 lastUpdateTimestamp_ = _minterStates[minter_].updateTimestamp;
+        MinterState storage minterState_ = _minterStates[minter_];
 
-        // MinterGateway already has more recent collateral update
-        if (newTimestamp_ <= lastUpdateTimestamp_) revert StaleCollateralUpdate(newTimestamp_, lastUpdateTimestamp_);
+        // The earliest allowed timestamp for a collateral update is the maximum of the last update timestamp and the latest proposed retrieval timestamp.
+        uint40 earliestAllowedTimestamp_ = UIntMath.max40(
```

Comment:
moving to max(a, b, c)  with freshness check from another pr. can belong in this pr as well.

Reply:
In my opinion, it is unrelated, and clearer to be done in steps.